lib/services/nodes-api-service.js

```diff
@@ -0,0 +1,132 @@
+// Copyright 2015, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+
+module.exports = nodeApiServiceFactory;
+di.annotate(nodeApiServiceFactory, new di.Provide('Http.Services.Api.Nodes'));
+di.annotate(nodeApiServiceFactory,
+    new di.Inject(
+        'Protocol.TaskGraphRunner',
+        'Services.Waterline',
+        'Errors',
+        'Logger',
+        '_',
+        'Promise'
+    )
+);
+function nodeApiServiceFactory(
+    taskGraphProtocol,
+    waterline,
+    Errors,
+    Logger,
+    _,
+    Promise
+) {
+    var logger = Logger.initialize(nodeApiServiceFactory);
+
+    function NodeApiService() {
+    }
+
+    /**
+     * Find Enclosure nodes which enclose compute node
+     * @param  {Object}     node
+     * @return {Promise}
+     */
+    NodeApiService.prototype._findEnclNodes = function(node) {
+        // Find the enclosure nodes who enclose this compute node
+        if (!_.has(node, 'relations')) {
+            return Promise.resolve();
+        }
+
+        var relation = _.find(node.relations, { relationType: 'enclosedBy' });
+        if (!relation || !_.has(relation, 'targets') ) {
+            return Promise.resolve();
+        }
+
+        return Promise.map(relation.targets, function (enclNodeId) {
+            return waterline.nodes.needByIdentifier(enclNodeId)
+            .catch(function (err) {
+                logger.error("Error Getting Enclosure Node", { error: err });
+                return;
+            });
+        });
+    };
+
+    /**
+     * Remove the relations between node and enclosure node, if enclosure node
+     * doesn't enclose any nodes, this enclosure node is also removed.
+     * @param  {Object}     node
+     * @param  {Array}      enclNodes
+     * @return {Promise}
+     */
+    NodeApiService.prototype._removeRelations = function(node, enclNodes) {
```

Comment:
not big problem. the better name is "removeEnclosureRelations" because it cannot be used to remove other future relations.